Fix msys2 build

mikewl opened the pull request:
This fixes msys2 building for the most part. All tests apart from libgit2 pass. Distributed also errors.

The toolchain used to build the test was actually the mingw64 from msys2. I am unsure what other issues are caused by using that but what is nice is that it allowed for the setup step to just be:
```bash
pacman -Syu
pacman -Su
pacman -S diffutils git m4 make patch tar p7zip msys/openssh ca-certificates pkg-config mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake mingw-w64-x86_64-ninja
```
Then open the mingw64 terminal and clone and make. 
`mbedtls` only seems to build when using ninja for some reason which is why ninja is now required for the msys2 build. Libgit2 is compiled with `USE_SSH=OFF` as compiling it with the libssh2 compiled causes the dll generated to not work with Julia for some reason. `ldd` in msys2 outputs question marks while `ldd` in Cygwin gives a perfectly fine output.

tkelman commented on deps/tools/git-external.mk:
```diff
@@ -53,7 +53,21 @@ $5/$1/source-compiled: $5/$1/.git/HEAD
+	$$(JLCHECKSUM) $$<
+	-rm -r $$(dir $$@)
+	mkdir -p $$(dir $$@)
+	-$(TAR) -C $$(dir $$@) --strip-components 1 -xf $$<
```
if this is the only line that's different, isn't this the only line that needs to be added?

Mikewl replied:
It was one of the first few edits I made and `ifeq` didn't work inside the block. Committed a better version that just adds the necessary portion.

tkelman commented on deps/tools/common.mk:
```diff
@@ -27,7 +27,11 @@ ifneq ($(VERBOSE), 0)
 endif
 # The call to which here is to work around https://cmake.org/Bug/view.php?id=14366
+ifeq ($(BUILD_OS),WINNT)
+CMAKE_COMMON += -DCMAKE_C_COMPILER="$(CC_BASE)"
```
why don't absolute paths work?

Mikewl replied:
the file extension is missing. `$$(which $(CC_BASE))` returns `C:/msys64/mingw64/bin/gcc` which doesn't include the file extension. The referenced bug has also been fixed since cmake >= 3.0 which is available on both Cygwin and windows.
Alternatively I can change it to `CMAKE_COMMON += -DCMAKE_C_COMPILER="$$(which $(CC_BASE)).exe"` which works as well.

tkelman replied:
maybe we could drop the workaround entirely since llvm requires cmake 3.4 anyway

Mikewl replied:
According to the listed cmake bug it was fixed in 3.0 so that should be perfectly fine.

tkelman commented on deps/libgit2.mk:
```diff
@@ -28,6 +28,7 @@ endif
 endif
 ifeq ($(BUILD_OS),WINNT)
 LIBGIT2_OPTS += -G"MSYS Makefiles"
+LIBGIT2_OPTS += -DBUILD_CLAR=OFF -DUSE_SSH=OFF
```
don't think we should be disabling ssh support

Mikewl replied:
https://gist.github.com/Mikewl/9c644213ed1aa221c121b7ca201e629b

If `USE_SSH` is enabled then ldd returns the above result. Disabled, below. The above result is unusable in Julia even though in Cygwin it returns a perfectly valid result (but not in msys2).

tkelman replied:
do we know where it's going wrong?

Mikewl replied:
I have looked but it’s beyond me unfortunately.

tkelman commented on deps/tools/git-external.mk:
```diff
@@ -63,20 +63,8 @@ $5/$$($2_SRC_DIR)/source-extracted: $$($2_SRC_FILE)
-	$(TAR) -C $$(dir $$@) --strip-components 1 -xf $$<
-	echo 1 > $$@
-
-else
```
should drop the `ifeq ($(BUILD_OS),WINNT)` above and probably move the comment

Mikewl replied:
Fixed. Should have tested in Cygwin first. Removed the comment rather.

mikewl committed:
Fix msys2 build

Target the new tar command more efficiently

helps if you remove the extra if too

reduce further

tkelman commented:
> `mbedtls` only seems to build when using ninja for some reason

I remember seeing this go wrong last time I tested in msys2, but that was long enough ago that I don't remember the details. Is it something we should attempt to fix in mbedtls' cmakelists? Or would the Unix Makefiles generator work maybe?

Mikewl commented:
```
-- Found Perl: C:/msys64/usr/bin/perl.exe (found version "5.24.1")
CMake Error at tests/CMakeLists.txt:126 (message):
  Could not create symbolic link for:
  C:/julia/deps/srccache/mbedtls-2.5.1-gpl/tests/data_files --> Parameter
  format not correct - "julia".



-- Configuring incomplete, errors occurred!
See also "C:/julia/deps/scratch/mbedtls-2.5.1-gpl/CMakeFiles/CMakeOutput.log".
make: *** [/c/julia/deps/mbedtls.mk:34: scratch/mbedtls-2.5.1-gpl/build-configured] Error 1
make: Leaving directory '/c/julia/deps'
```
There's the error. It has something to do with the perl script for tests. Disabling mbedtls tests allows for msys makefiles to be used for building, removing the need for ninja. Not sure which is preferred so I will leave it as is for now.

tkelman commented:
Do the mbedtls tests run if you build them with ninja? If not I'd prefer disabling the tests when built in msys2, but not too strongly. Or maybe patch the cmakelists to make copies instead of symlinks when building in msys, but haven't looked at exactly what it's doing.

Mikewl commented:
Well, 60/60 tests fail with `OTHER_FAULT` with the ninja build.
So, should I rather just disable tests and leave it as `MSYS Makefiles`?
